### PR TITLE
Add example snap for `snap_getLocale`

### PR DESCRIPTION
**Pull request by FrederikBolding**
Adds an example snap for `snap_getLocale` and a test-snaps section to use in E2E.

Fixes https://github.com/MetaMask/snaps/issues/1682

### CODE REVIEW

File: packages/examples/packages/get-locale/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  }
+
+  return translatedString;
+};
+
+/**
+ * Handle incoming JSON-RPC requests from the dapp, sent through the
+ * `wallet_invokeSnap` method. This handler handles a single method:
+ *
+ * - `hello`: Shows an alert dialog with a localized message.
```
Comment (Mrtenz) [inline packages/examples/packages/get-locale/src/index.ts:49]:
```suggestion
 * - `hello`: Shows an alert dialog with a localized message. This uses the
 * `snap_getLocale` JSON-RPC method to get the locale of the user's MetaMask
 * instance.
```

File: packages/examples/packages/get-locale/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
+import { rpcErrors } from '@metamask/rpc-errors';
+import type { OnRpcRequestHandler } from '@metamask/snaps-types';
+import { DialogType } from '@metamask/snaps-types';
+import { panel, text, heading } from '@metamask/snaps-ui';
+import { assert } from '@metamask/utils';
+
+const FALLBACK_LANGUAGE = 'en';
+
+const TRANSLATIONS: Record<string, Record<string, string>> = {
```
Comment (Mrtenz) [inline packages/examples/packages/get-locale/src/index.ts:9]: I would add some comments explaining these variables.

File: packages/examples/packages/get-locale/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    da: 'Hej $1!',
+  },
+  greetingBody: {
+    en: 'This is a dialog!',
+    da: 'Dette er en dialog!',
+  },
+};
+
+/**
+ * Provides basic translation given a locale, a translation key and some optional arguments.
```
Comment (Mrtenz) [inline packages/examples/packages/get-locale/src/index.ts:21]:
```suggestion
 * Get a translated message given a locale, a translation key, and some optional
 * arguments. The arguments replace the placeholders in the translation.
 *
 * In a real world scenario, this function would likely be implemented by some
 * translation library, but for the sake of simplicity, we're just using a
 * simple function here.
```

File: packages/examples/packages/get-locale/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+ *
+ * @param locale - The currently selected user locale.
+ * @param translationKey - The translation key to translate.
```
Comment (Mrtenz) [inline packages/examples/packages/get-locale/src/index.ts:24]:
```suggestion
 * @param translationKey - The key of the message in {@link TRANSLATIONS}
 * to translate.
```

File: packages/examples/packages/get-locale/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+ * @param args - An optional array of arguments for the translation.
+ * @returns The translated string.
+ * @throws If the `translationKey` does not exist.
+ */
+const translate = (locale: string, translationKey: string, args?: string[]) => {
+  const translationsForKey = TRANSLATIONS[translationKey];
+  assert(translationsForKey, `Invalid translation key: ${translationKey}`);
```
Comment (Mrtenz) [inline packages/examples/packages/get-locale/src/index.ts:31]:
```suggestion
  assert(translationsForKey, `Unknown translation key: ${translationKey}.`);

```

**Review by Mrtenz — CHANGES_REQUESTED** [review]

File: packages/examples/packages/get-locale/src/index.ts (hunk 2 above)
Comment (FrederikBolding) [reply packages/examples/packages/get-locale/src/index.ts]: https://github.com/MetaMask/snaps/pull/1684/commits/24f5bc354162a7fb9aa85713d97c33b2b5037bc5

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**